weave/flow/eval_imperative.py

```diff
@@ -0,0 +1,371 @@
+
+            # --- Setup the evaluation object ---
+            @weave.op(name="Evaluation.evaluate")
+            def evaluate(self: Evaluation, model: Model) -> None: ...
+
+            @weave.op(name="Evaluation.predict_and_score")
+            def predict_and_score(
+                self: Evaluation, model: Model, example: dict
+            ) -> dict:
+                predict_method = cast(Op, model.get_infer_method())
+                model_output, predict_call = predict_method.call(model, example)
+                current_predict_call.set(predict_call)
+
+                # This data is just a placeholder to give a sense of the data shape.
+                # The actual output is explicitly replaced in ImperativeScoreLogger.finish.
+                return {
+                    "model_output": model_output,
+                    "scores": {},
+                    "model_latency": None,
+                }
+
+            @weave.op(name="Evaluation.summarize")
+            def summarize(self: Evaluation) -> dict:
+                return cast(dict, current_summary.get())
+
+            self._pseudo_evaluation.__dict__.update(
+                {
+                    "evaluate": MethodType(evaluate, self._pseudo_evaluation),
+                    "predict_and_score": MethodType(
+                        predict_and_score, self._pseudo_evaluation
+                    ),
+                    "summarize": MethodType(summarize, self._pseudo_evaluation),
+                }
+            )
+
+            # Create the evaluation call
+            wc = require_weave_client()
+            self._evaluate_call = wc.create_call(
+                display_name=default_evaluation_display_name,
+                op=self._pseudo_evaluation.evaluate,
+                inputs={
+                    "self": self._pseudo_evaluation,
+                    "model": self.model,
+                },
+            )
+            assert self._evaluate_call is not None
+            call_context.push_call(self._evaluate_call)
+
+        # Make the prediction call
+        with _set_current_output(output):
+            _, predict_and_score_call = self._pseudo_evaluation.predict_and_score.call(
+                self._pseudo_evaluation,
+                self.model,
+                inputs,
+                __require_explicit_finish=True,
+            )
+
+        # Get the predict_call from the context variable
+        predict_call = current_predict_call.get()
+        if predict_call is None:
+            raise ValueError("predict_call should not be None")
+
+        assert self._evaluate_call is not None
+        pred = ImperativeScoreLogger(
+            predict_and_score_call=predict_and_score_call,
+            evaluate_call=self._evaluate_call,
+            predict_call=predict_call,
+        )
+        self._accumulated_predictions.append(pred)
+        return pred
+
+    def log_summary(self, summary: dict) -> None:
```

Comment:
FUTURE: since we are in the business of accumulating, we could accumulate the scores and use our existing auto-summary to save the user from the hassle of summaries

Reply:
hmm yes, I guess we've gone this far we can accumulate for the user.

Agree to defer to a future PR